### PR TITLE
Terraform rework main workflow

### PR DESCRIPTION
**Pull request by YakMan101**
The terraform has been reworked to follow the new architecture diagram:
image

As well as provision the EC2 for streamlit hosting:
image

Terraform README.md has be updated accordingly.

ECS Tasks now initiate as soon as a CSV is added to the S3 bucket. The ECS tasks are able to run concurrently, they do not need to wait for each other.

Changes have also been made to the sentiment analyser Dockerfile to use a lower python version to fix the psycopg2 import bug.

### CODE REVIEW

File: dashboard/dashboard.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        if not scatter_data.empty:
+
+            scatter_chart = alt.Chart(scatter_data).mark_point(filled=True).encode(
+                x=alt.X('title_polarity_score:Q',
+                        scale=alt.Scale(domain=[-1, 1]),
+                        axis=alt.Axis(title='Title Polarity Score', grid=True)),
+                y=alt.Y('content_polarity_score:Q',
+                        scale=alt.Scale(domain=[-1, 1]),
+                        axis=alt.Axis(title='Content Polarity Score', grid=True)),
+                color=alt.Color('source_name:N', title='Source'),
```
Comment (megansam5) [inline dashboard/dashboard.py:101]: Wow what a sexy dashboard!!

File: dashboard/upload_dashboard.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
+source .env
```
Comment (megansam5) [inline dashboard/upload_dashboard.sh:1]: Very useful shell script!

File: news_sentiment_analyser/Dockerfile [hunk 3]
```diff
@@ -1,5 +1,5 @@
 # Specifies latest image of python
-FROM python:latest
+FROM python:3.11
```
Comment (megansam5) [inline news_sentiment_analyser/Dockerfile:2]: well done for finding a work around!

File: terraform/main.tf [hunk 4]
```diff
@@ -363,4 +323,202 @@ resource "aws_scheduler_schedule" "daily_step_function_trigger" {
     role_arn = aws_iam_role.step_functions_scheduler_role.arn
     input   = jsonencode({})
   }
+}
+
+# =========================== S3 Event to EventBridge ===========================
```
Comment (megansam5) [inline terraform/main.tf:328]: Well done doing the scheduling!!

File: dashboard/upload_dashboard.sh
Comment (indiah444) [inline dashboard/upload_dashboard.sh:1]: Really great idea to automate processes with shell scripts!

**Review by megansam5 — APPROVED** [review]
Thanks for doing this Shayak, we would wither without you!!

**Pull request merged**